src/compas_fab/ghpython/components/ros_connect.py

```diff
@@ -0,0 +1,96 @@
+import Grasshopper
+import System
+from ghpythonlib.componentbase import dotnetcompiledcomponent as component
+from scriptcontext import sticky as st
+
+from compas_fab.backends import RosClient
+from compas_fab.ghpython.components import create_id
+from compas_fab.ghpython.components.icons import ros_connect_icon
+
+
+class ROSConnect(component):
+    def __new__(cls):
+        instance = Grasshopper.Kernel.GH_Component.__new__(cls,
+                                                           "ROS Connect",
+                                                           "ROS Connect",
+                                                           """Connect or disconnect to ROS""",
+                                                           "COMPAS FAB",
+                                                           "ROS")
+        return instance
+
+    def get_ComponentGuid(self):
+        return System.Guid("cdd47086-f902-4b77-825b-6b79c3aaecc1")
+
+    def SetUpParam(self, p, name, nickname, description):
+        p.Name = name
+        p.NickName = nickname
+        p.Description = description
+        p.Optional = True
+
+    def RegisterInputParams(self, pManager):
+
+        p = Grasshopper.Kernel.Parameters.Param_String()
+        self.SetUpParam(p, "ip", "ip", "The ip address of ROS master. Defaults to 127.0.0.1")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_Integer()
+        self.SetUpParam(p, "port", "port", "The port of ROS master. Defaults to 9090.")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_Boolean()
+        self.SetUpParam(p, "connect", "connect", "If `True`, connect to ROS. Defaults to False.")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_Boolean()
+        self.SetUpParam(p, "disconnect", "disconnect", "If `True`, disconnect from ROS. Defaults to False.")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
```

Comment:
Would it make sense to simplify this input into just one boolean that is intended to use with the toggle component? So, if `True` we connect, if `False`, disconnect?

Reply:
agree, updated